From a093bcbaea39c650e921a6ce9c1fdb01aa487ad4 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Wed, 28 Oct 2026 13:50:05 +0000
Subject: [PATCH 275/277] addons: version the web app manifest as kodi.webapp

A web app declares a start URL, user agents, user scripts and content
filters, and what the web player does with them may change between Kodi
releases. kodi.webapp is a system add-on carrying that manifest's
version, so a web app imports it like a Python add-on imports
xbmc.python, and Kodi only offers a repository's web apps whose import
it can satisfy.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 addons/kodi.webapp/addon.xml        | 7 +++++++
 cmake/installdata/common/addons.txt | 1 +
 system/addon-manifest.xml           | 1 +
 3 files changed, 9 insertions(+)
 create mode 100644 addons/kodi.webapp/addon.xml

--- /dev/null
+++ b/addons/kodi.webapp/addon.xml
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon id="kodi.webapp" version="1.0.0" provider-name="Team Kodi">
+  <backwards-compatibility abi="1.0.0"/>
+  <requires>
+    <import addon="xbmc.core" version="0.1.0"/>
+  </requires>
+</addon>
--- a/cmake/installdata/common/addons.txt
+++ b/cmake/installdata/common/addons.txt
@@ -11,6 +11,7 @@ addons/kodi.inputstream/*
 addons/kodi.peripheral/*
 addons/kodi.resource/*
 addons/kodi.vfs/*
+addons/kodi.webapp/*
 addons/metadata.album.universal/*
 addons/metadata.artists.universal/*
 addons/metadata.common.allmusic.com/*
--- a/system/addon-manifest.xml
+++ b/system/addon-manifest.xml
@@ -25,6 +25,7 @@
   <addon>kodi.binary.instance.videocodec</addon>
   <addon>kodi.binary.instance.visualization</addon>
   <addon>kodi.resource</addon>
+  <addon>kodi.webapp</addon>
   <addon>metadata.album.universal</addon>
   <addon>metadata.artists.universal</addon>
   <addon>metadata.common.allmusic.com</addon>
